File: gen_mappings.rb
require 'cli'
require_relative 'lib/scenario_wiremock_template'
require_relative 'colorize'

def json_from_file(path)
  JSON.parse File.read(path)
end

CHECK = '✓'.green

settings = CLI.new do
  argument :scenario_path, description: 'Path of the JSON file containing scenarios'
  argument :output_path, description: 'Path to store the wiremock mapping files'
end.parse!

summary = []

scenarios = json_from_file(settings.scenario_path)

wiremock_stubs = ScenarioTemplates
  .new(scenarios)
  .as_wiremock_stubs
  .to_h

wiremock_stubs.each_pair do |name, scenario|
  File.open(settings.output_path + name + '.json', 'w') do |f|
    f.write(scenario)
    summary.push "#{CHECK}  #{name}"
  end
end

puts 'Summary:'
puts summary.join("\n")

File: gen_mappings.js
var fs = require('fs');
var _ = require('underscore');
var colors = require('colors/safe');

var scenarioMappings = require('./lib/scenario_mappings');

var argv = require('yargs')
    .alias('s', 'scenarios')
    .describe('s', 'Path of JSON scenarios file')
    .alias('o', 'output_dir')
    .describe('o', 'Directory to wiremock mapping files')
    .demandOption(['scenarios', 'output_dir'])
    .argv;

var scenarios = JSON.parse(fs.readFileSync(argv.scenarios));

var wiremockStubs = scenarioMappings.buildStubsFromScenarios(scenarios);

_.each(wiremockStubs, function(stub) {
    fs.writeFile(argv.output_dir + stub.name + '.json', JSON.stringify(stub.mapping, null, 2), printResult(stub.name));
});

function printResult(name) {
    return function(error, result) {
        if (error) {
            var cross = colors.red("✗");
            console.log(cross + " " + name);
        }
        else {
            var check = colors.green("✓");
            console.log(check + " " + name);
        }
    };
}
